waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/DatabaseWhitelistAccessControlHandler.java

```diff
@@ -41,15 +45,36 @@ public DatabaseWhitelistAccessControlHandler(PrimaryMetaStore primaryMetaStore,
   }
 
   private void add(String databaseName) {
-    databaseWhiteList.add(databaseName.trim().toLowerCase());
+    databaseWhiteList.add(trimToLowerCase(databaseName));
+  }
+
+  private String trimToLowerCase(String string) {
+    return string.trim().toLowerCase();
   }
 
   @Override
   public boolean hasWritePermission(String databaseName) {
     if (databaseName == null) {
       return true;
     }
-    return databaseWhiteList.contains(databaseName.trim().toLowerCase());
+    databaseName = trimToLowerCase(databaseName);
+    if (databaseWhiteList.contains(databaseName) || databaseWhiteList.contains("*")) {
+      return true;
+    }
+
+    char firstLetterOfDatabaseName = databaseName.charAt(0);
```

Comment:
Aah, I guess this is back to "WIP" now seeing as we decided on full regex right? So I'll stop reviewing for now...